src/mol-repr/structure/visual/util/color.ts

```diff
@@ -0,0 +1,106 @@
+/**
+ * Copyright (c) 2021 mol* contributors, licensed under MIT, See LICENSE file for more info.
+ *
+ * @author Alexander Rose <alexander.rose@weirdbyte.de>
+ */
+
+import { calcMeshColorSmoothing } from '../../../../mol-geo/geometry/mesh/color-smoothing';
+import { calcTextureMeshColorSmoothing } from '../../../../mol-geo/geometry/texture-mesh/color-smoothing';
+import { MeshValues } from '../../../../mol-gl/renderable/mesh';
+import { TextureMeshValues } from '../../../../mol-gl/renderable/texture-mesh';
+import { WebGLContext } from '../../../../mol-gl/webgl/context';
+import { Texture } from '../../../../mol-gl/webgl/texture';
+import { Theme } from '../../../../mol-theme/theme';
+import { ValueCell } from '../../../../mol-util';
+import { ParamDefinition as PD } from '../../../../mol-util/param-definition';
+
+export const ColorSmoothingParams = {
+    smoothColors: PD.MappedStatic('auto', {
+        auto: PD.Group({}),
+        on: PD.Group({
+            resolutionFactor: PD.Numeric(2, { min: 1, max: 6, step: 0.1 }),
+            sampleStride: PD.Numeric(6, { min: 1, max: 12, step: 1 }),
+        }),
+        off: PD.Group({})
+    }),
+};
+export type ColorSmoothingParams = typeof ColorSmoothingParams
+
+export function getColorSmoothingProps(props: PD.Values<ColorSmoothingParams>, theme: Theme, resolution?: number, webgl?: WebGLContext) {
+    if ((props.smoothColors.name === 'on' || (props.smoothColors.name === 'auto' && theme.color.preferSmoothing)) && resolution && resolution < 3 && webgl) {
+        let stride = 6;
+        if (props.smoothColors.name === 'on') {
+            resolution *= props.smoothColors.params.resolutionFactor;
+            stride = props.smoothColors.params.sampleStride;
+        } else {
+            if (resolution > 0.5 && resolution < 1) {
+                resolution *= 2 * resolution;
```

Comment:
Why ``v <- 2 * v * v``  for values between 0.5 and 1? This seems quite random and there is a weird discontinuity with the other options.

E.g. if resolution = 0.5, the final value will be 1, but if resolution = 0.500001, it will end up being ~0.5 (0.5 * 0.5 * 2).

Reply:
Yeah, that needed some more work. Changed it to https://graphtoy.com/?f1(x,t)=(2-smoothstep(0,1.1,x))*x&coords=0.7,0.6,1.8